src/csvcubed/models/sparqlresults.py

```diff
@@ -438,75 +449,77 @@ def _map_qube_component_sparql_result(
         required=none_or_map(result_dict.get("required"), bool) or False,
     )
     return result
 
 
-def _map_obs_val_for_dsd_component_properties_results(
-    sparql_results: List[ResultRow],
-) -> List[ObsValDsdComponentResult]:
-    def map_row(row_result: Dict[str, Any]) -> ObsValDsdComponentResult:
-        return ObsValDsdComponentResult(
-            csv_column_property_url=none_or_map(
-                row_result.get("csvColumnPropertyUrl"), str
-            ),
-            observation_value_column_titles=none_or_map(
-                row_result.get("observationValueColumnTitles"), str
-            ),
-            dsd_uri=str(row_result["dsd_uri"]),
-        )
-
-    return [map_row(row.asdict()) for row in sparql_results]
-
-
 def map_qube_components_sparql_result(
     sparql_results_dsd_components: List[ResultRow],
-    sparql_results_obs_val_col_titles: List[ResultRow],
     json_path: Path,
-) -> QubeComponentsResult:
+    map_dsd_uri_to_csv_url: Dict[str, str],
+    map_csv_url_to_column_definitions: Dict[str, List[ColumnDefinition]],
+) -> Dict[str, QubeComponentsResult]:
     """
-    Maps sparql query result to `QubeComponentsResult`
+    Returns a map of csv_url to `QubeComponentsResult`
 
     Member of :file:`./models/sparqlresults.py`
 
-    :return: `QubeComponentsResult`
+    :return: `Dict[str, QubeComponentsResult]`
     """
-    obs_val_col_title_results: List[
-        ObsValDsdComponentResult
-    ] = _map_obs_val_for_dsd_component_properties_results(
-        sparql_results_obs_val_col_titles
-    )
+    components: List[QubeComponentResult] = [
+        _map_qube_component_sparql_result(r, json_path)
+        for r in sparql_results_dsd_components
+    ]
 
-    components: List[QubeComponentResult] = []
-    for result in sparql_results_dsd_components:
-        dsd_component_result: QubeComponentResult = _map_qube_component_sparql_result(
-            result, json_path
-        )
-        if any(obs_val_col_title_results):
-            obs_val_col_title_result_for_component = [
-                obs_val_col_title_result
-                for obs_val_col_title_result in obs_val_col_title_results
-                if obs_val_col_title_result.csv_column_property_url
-                == dsd_component_result.property
-                and obs_val_col_title_result.dsd_uri == dsd_component_result.dsd_uri
+    map_dsd_uri_to_components = group_by(components, lambda c: c.dsd_uri)
+
+    for (dsd_uri, components) in map_dsd_uri_to_components.items():
+        csv_url = map_dsd_uri_to_csv_url[dsd_uri]
+        csv_column_definitions = map_csv_url_to_column_definitions[csv_url]
+
+        measure_uris = {
+            c.property for c in components if c.property_type == str(QB.MeasureProperty)
+        }
+
+        observed_value_columns = [
+            c
+            for c in csv_column_definitions
+            if (not c.virtual) and c.property_url in measure_uris
```

Comment:
I don't think this matching is working at the moment. When debugging, I saw he `observed_value_columns` empty, when it needs to have some values. I think we need to do `startsWith` checking or similar.